hwtypes/smt_utils.py

```diff
@@ -0,0 +1,76 @@
+from . import SMTBit
+import operator
+from functools import partial, reduce
+import abc
+
+_or_reduce = partial(reduce, operator.or_)
+_and_reduce = partial(reduce, operator.and_)
+
+class FormulaConstructor:
+    @abc.abstractmethod
+    def serialize(self, ts, indent):
```

Comment:
I would prefer a visitor pattern with serialization visitor but I can live with this,